### PR TITLE
fix(learn): rework Rosetta Code SHA256

### PR DESCRIPTION
**Pull request by Sembauke**
- Removes unnecessary links;
- Gives a more robust description of the challenge.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

ref: #40896

### CODE REVIEW

**Review by gikf — COMMENTED** [review]
Link here has one advantage - it gives some leads how on exactly `SHA-256` is computed. Without putting that information in the description, I'd consider keeping some link. If not to wikipedia, then just to specification for implementation (https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf). Or maybe some dedicated article with that?

**Comment by Sembauke** [comment]
www.freecodecamp.org/news/md5-vs-sha-1-vs-sha-2-which-is-the-most-secure-encryption-hash-and-how-to-check-them/amp/  This seems like it could do the trick

**Comment by moT01** [comment]
There some [related discussion here](https://github.com/freeCodeCamp/freeCodeCamp/issues/40749). Is there any way to write enough instructions that users wouldn't need a link to solve this? I think that would be ideal.

**Comment by Sembauke** [comment]
> There's some [related discussion here](https://github.com/freeCodeCamp/freeCodeCamp/issues/40749). Is there any way to write enough instructions that users wouldn't need a link to solve this? I think that would be ideal.

Yep, that is what I based this refactor on, you should not need a link to solve this challenge, to be honest. Just giving it a bit more description should be enough.

**Comment by moT01** [comment]
So, this is ready to go then?

**Comment by Sembauke** [comment]
> So, this is ready to go then?

Yes

**Comment by gikf** [comment]
I still think it'd be good to include link to implementation specification, as that's after all what is required to complete this challenge. Alternatively maybe some kind of explicit note should be included, that it is not expected to come up with how SHA is hashing, but to *just* implement it, after finding details with own research.

**Comment by Sembauke** [comment]
@gikf  What would you suggest what the note would look like?

**Review by gikf — DISMISSED** [review]
LGTM :+1:

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sha-256.md [hunk 1]
```diff
@@ -8,11 +8,12 @@ dashedName: sha-256
...
 # --description--
 
-**[SHA-256](https://en.wikipedia.org/wiki/SHA-256)** is the recommended stronger alternative to [SHA-1](https://rosettacode.org/wiki/SHA-1).
+The `SHA-2` family is a stronger alternative to `SHA-1`. The main difference between them is the length of the hash.
+Meaning `SHA-1` provides a shorter code with fewer possibilities for unique combinations. `SHA-2` or `SHA-256` creates a longer and thus more complex hash with more possibilities.
 
 # --instructions--
 
-Write a function that takes a string as a parameter and returns its SHA-256 digest.
+Research implemenation details and write a function that takes a string as the parameter and make it return a hash using `SHA-256`
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sha-256.md:16]:
```suggestion
Research implemenation details and write a function that takes a string as the parameter and returns a hash using `SHA-256`
```

I think we should stick with the "function... takes... returns..." syntax here. Thoughts?

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sha-256.md (hunk 1 above)
Comment (Sembauke) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sha-256.md:16]: Seems good to me 👍

**Review by gikf — APPROVED** [review]

**Review by naomi-lgbt — APPROVED** [review]
I'm good with this now. Nice work Sem. :)

**Pull request merged**